### PR TITLE
Fix m1 signatures

### PR DESCRIPTION
**Pull request by ameily**
This PR attempts to fix the Mac signatures and associated tests for M1 architecture.

- Added the `arm64` architecture to the signature check list.
- Created a small unsigned binary to test against on M1 systems.
- Make additional changes to a signed test program to trigger the invalid signature test.

Refs osquery/osquery#7359

### CODE REVIEW

**Comment by ameily** [comment]
The invalid signature test is still failing for arm64 architectures. The test itself is labeled as a hack because it copies a signed binary, `ls`, and modifies several bytes, hopefully triggering an invalid signature. The problem appears to be that the arm64 code section is not being modified so the arm64 code section is always valid. I tried bumping up the number of bytes being modified which eventually corrupts the entire signature, not just the arm64 code section.

I've verified the results against `codesign`, so the test itself is working properly.

**Comment by ameily** [comment]
I think I figured it out. I made it so that the `test_get_invalid_signature` test actually modifies one byte in the middle of *each* architecture slice. This may be overkill, but, assuming that the structure for universal binaries doesn't change, should work as new architectures are added.

File: osquery/tables/system/tests/CMakeLists.txt [hunk 1]
```diff
@@ -135,6 +135,13 @@ function(generateOsqueryTablesSystemDarwinTests)
     tests_helper
     thirdparty_googletest
   )
+
+  add_executable(unsigned_test darwin/unsigned_test.cpp)
```
Comment (alessandrogario) [inline osquery/tables/system/tests/CMakeLists.txt:139]: In the upcoming CI changes, we'll package all the test binaries and send them to the Catalina worker; can this be changed to something like this (warning: pseudocode)?

```cmake
add_executable(unsigned_test ..)

add_custom_command(
  OUTPUT "${TEST_CONFIGS_DIR}/unsigned_test"
  DEPENDS unsigned_test
  COMMAND "${CMAKE_COMMAND}" -E $<TARGET_FILE:unsigned_test> "${TEST_CONFIGS_DIR}/unsigned_test_temp"
  COMMAND codesign ... "${TEST_CONFIGS_DIR}/unsigned_test_temp"
  COMMAND "${CMAKE_COMMAND}" -E rename  "${TEST_CONFIGS_DIR}/unsigned_test_temp" "${TEST_CONFIGS_DIR}/unsigned_test"
  VERBATIM
  COMMENT "Generating unsigned_test..."
)

add_custom_target(unsigned_test_builder
  DEPENDS "${TEST_CONFIGS_DIR}/unsigned_test"
)

# Make sure that the data directory is generated before the unsigned_test data
add_dependencies(unsigned_test_builder
  osquery_tools_test_configfiles
)

# Make sure that the data is built before the test
add_dependencies(osquery_tables_system_darwin_tests-test
  unsigned_test_builder
)
```

**Review by alessandrogario — CHANGES_REQUESTED** [review]

**Comment by ameily** [comment]
@alessandrogario I believe I addressed your cmake feedback and updated the test to use the new output location, `test_configs/unsigned_test`

File: osquery/tables/system/tests/darwin/unsigned_test.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * used in the SignatureTest.test_get_unsigned unit test. The cmake post build
+ * will remove any ad hoc signature applied on M1 systems.
+ */
+
+#include <stdio.h>
+
+int main(int argc, char** argv) {
+  printf("Hello world!\n");
+  return 0;
+}
```
Comment (mike-myers-tob) [inline osquery/tables/system/tests/darwin/unsigned_test.cpp]: nitpick: end this file with a newline

File: tools/ci/scripts/package_tests.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  local source="$1"
+  local destination="$(realpath $2)"
+
+  printf "Source path: ${source}\n"
+  printf "Destination path: ${destination}\n\n"
+
+  ( cd "${source}" && copy_files "${destination}" ) || return 1
+  patch_ctest_files "$(realpath ${source})" "${destination}" || return 1
+
+  printf "Generating the launcher..\n"
```
Comment (mike-myers-tob) [inline tools/ci/scripts/package_tests.sh]: Nitpick: some of the printf statement strings end in a `..` instead of a `...`

**Comment by mike-myers-tob** [comment]
Ok, I verified that it builds and passes tests on macOS 12 (x86 VM). I like the improvements to the formerly "hacky" signature invalidation method during the signature tests.

Code looks good, just need @alessandrogario to confirm if the conflicted CI file can be resolved so we can merge this PR.

**Comment by ameily** [comment]
@mike-myers-tob I rebased from `feature/m1-support` which got rid of the CI conflict and addressed your other feedback.

**Pull request merged**